vowpalwabbit/core/src/reductions/interaction_ground.cc

```diff
@@ -3,111 +3,240 @@
+  std::swap(destination->data.ftrl_alpha , source->data.ftrl_alpha);
+  std::swap(destination->data.ftrl_beta , source->data.ftrl_beta);
+  std::swap(destination->data.l1_lambda , source->data.l1_lambda);
+  std::swap(destination->data.l2_lambda , source->data.l2_lambda);
+  std::swap(destination->data.predict , source->data.predict);
+  std::swap(destination->data.normalized_squared_norm_x , source->data.normalized_squared_norm_x);
+  std::swap(destination->data.average_squared_norm_x , source->data.average_squared_norm_x);
+  std::swap(destination->no_win_counter , source->no_win_counter);
+  std::swap(destination->early_stop_thres , source->early_stop_thres);
+  std::swap(destination->ftrl_size , source->ftrl_size);
+  std::swap(destination->total_weight , source->total_weight);
+  std::swap(destination->normalized_sum_norm_x , source->normalized_sum_norm_x);
 }
-}  // namespace
 
 base_learner* VW::reductions::interaction_ground_setup(VW::setup_base_i& stack_builder)
 {
-  options_i& options = *stack_builder.get_options();
+  options_i& pi_options = *stack_builder.get_options();
+  VW::workspace *all = stack_builder.get_all_pointer();
   bool igl_option = false;
 
   option_group_definition new_options("[Reduction] Interaction Grounded Learning");
   new_options.add(make_option("experimental_igl", igl_option)
-                      .keep()
-                      .necessary()
-                      .help("Do Interaction Grounding with multiline action dependent features")
-                      .experimental());
+                  .keep()
+                  .necessary()
+                  .help("Do Interaction Grounding with multiline action dependent features")
+                  .experimental());
 
-  if (!options.add_parse_and_check_necessary(new_options)) { return nullptr; }
+  if (!pi_options.add_parse_and_check_necessary(new_options)) { return nullptr; }
 
-  // number of weight vectors needed
-  size_t problem_multiplier = 2;  // One for reward and one for loss
+  // 2 weight vectors needed. 0 for ik, 1 for pi
+  size_t problem_multiplier = 2;
   auto ld = VW::make_unique<interaction_ground>();
 
-  // Ensure cb_adf so we are reducing to something useful.
-  if (!options.was_supplied("cb_adf")) { options.insert("cb_adf", ""); }
+  ld->ik_ex = VW::alloc_examples(1);
+
+  // Ensure cb_explore_adf so we are reducing to something useful.
+  // Question: Can we support both cb_adf and cb_explore_adf?
+  if (!pi_options.was_supplied("cb_explore_adf")) {
+    pi_options.insert("cb_explore_adf", "");
+  }
+
+  multi_learner *pi_learner = as_multiline(stack_builder.setup_base_learner());
+
+  // 1. fetch already allocated coin reduction
+  auto* ftrl_coin = pi_learner->get_learner_by_name_prefix("ftrl-Coin");
+
+  // 2. prepare args for ik stack
+  std::string psi_args = "--quiet --link=logistic --loss_function=logistic --coin";
+
+  int argc = 0;
+  char** argv = to_argv_escaped(psi_args, argc);
```

Comment:
`to_argv_escaped` has many issues and should not be used. Please use `VW::split_command_line`